common/components/ui/Address.tsx

```diff
@@ -1,40 +1,46 @@
 import React from 'react';
-import { toChecksumAddress } from 'ethereumjs-util';
 import NewTabLink from './NewTabLink';
 import { IWallet } from 'libs/wallet';
 import { BlockExplorerConfig } from 'types/network';
+import { getChecksumAddressFn } from 'selectors/config';
+import { AppState } from 'reducers';
+import { connect } from 'react-redux';
 
 interface BaseProps {
   explorer?: BlockExplorerConfig | null;
+  address?: string | null;
+  wallet?: IWallet | null;
 }
 
-interface AddressProps extends BaseProps {
-  address: string;
+interface StateProps {
+  toChecksumAddress: ReturnType<typeof getChecksumAddressFn>;
 }
 
-interface WalletProps extends BaseProps {
-  wallet: IWallet;
-}
-
-type Props = AddressProps | WalletProps;
-
-const isAddressProps = (props: Props): props is AddressProps =>
-  typeof (props as AddressProps).address === 'string';
-
-const Address: React.SFC<Props> = props => {
-  let addr = '';
-  if (isAddressProps(props)) {
-    addr = props.address;
-  } else {
-    addr = props.wallet.getAddressString();
+type Props = BaseProps & StateProps;
+
+// const isAddressProps = (props: Props): props is Props =>
+//   typeof (props as AddressProps).address === 'string';
+
+export class Address extends React.PureComponent<Props> {
+  public render() {
+    const { wallet, address, explorer, toChecksumAddress } = this.props;
+    let addr = '';
+    if (address != null) {
```

Comment:
This should be using strict inequality

Reply:
Surprised the linter didn't bark at me for this.